Sync v2-6-stable with v2-6-test to release 2.6.1

ephraimbuddy opened the pull request:
Time for `2.6.1rc1`!

pierrejeambrun committed:
Present affected task instances as table (#30633)

(cherry picked from commit 90ee7639478bc8534164c4bbb05a1ccfb04272dd)

Bowrna committed:
docker compose doc changes (#30662)

(cherry picked from commit 1cf1fc46e95c889c8c81bb2ed2c3f08a681bf7e3)

Commit:
Reenable clear on TaskInstanceModelView for role User (#30415)

* Reenable clear on TaskInstanceModelView for role User

The action was disable in https://github.com/apache/airflow/pull/20659
which resolved https://github.com/apache/airflow/issues/20655. The issue
only mentions that the edit is broken and should be disabled. So it seem
like the disabling of the clear action was unintentional.

Also based on the discussion in the PR
https://github.com/apache/airflow/issues/20655 further reinforces this.
That the author believed it still worked could be explain by that using
a user with role `Admin` the action was still available and therefore
one could easily make a mistake believing it still worked as expected.

This PR reenables it action and modifies and existing test case to also
verify that clearing is possible using a user with the role `User`.

* Add back other set state actions

* fix static checks

---------

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>
(cherry picked from commit b140c4473335e4e157ff2db85148dd120c0ed893)

potiuk committed:
Update the user-facing documentation of providers (#30816)

We've recently clarified and described our policies for accepting
providers to be maintained by the community (#30657) - this was
directed towards the Airflow developers and contributors. This PR
reviews user-facing part of the documentation for providers by
removing some obsolete/not very useful documentation and pointing
to the new policy where appropriate.

(cherry picked from commit 76ebc9ba8a1ca26014f5142ac28035bef2702c27)

ZaneBartlett1 committed:
Improve Quick Start instructions (#30820)

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>
(cherry picked from commit f5f35b3b6bd67d7a3cfef207f9b3e72fe68033f1)

Commit:
Prevent DagRun's `start_date` from reset (#30124) (#30125)

(cherry picked from commit 070ecbd87c5ac067418b2814f554555da0a4f30c)

bbovenzi committed:
Fix mappedStates case (#30916)

* Fix mappedStates case

* remove stray console log

(cherry picked from commit decaaa3df2b3ef0124366033346dc21d62cff057)

bbovenzi committed:
Improve task group UI in new graph (#30918)

* Improve task group UI in new graph

* adjust task group color

(cherry picked from commit de0cdd77c4d25efd75be02dce50fd5e202fc8a23)

bbovenzi committed:
Include node height/width in center-on-task logic (#30924)

(cherry picked from commit d88ef477f7daf86d2ec9ad6650e75d00e5e78e36)

ephraimbuddy committed:
Some minor updates to the release command (#30944)

pygithub now requires to specify if a search is for pull request or issues and not both.
Also, a fix on searching old rc versions after removing the asf svn repo

(cherry picked from commit 3d6ad50745129e2fa9e97f5047ff1bfa784934a9)

pankajastro committed:
Remove print from dag trigger command (#30921)

(cherry picked from commit 36fe6d0377d37b5f6be8ea5659dcabb44b4fc233)

pankajastro committed:
Fix `order_by` request in list DAG rest api (#30926)

(cherry picked from commit 1d4b1410b027c667d4e2f51f488f98b166facf71)

LuizArmesto committed:
Use the Task Group explicitly passed to 'partial' if any (#30933)

(cherry picked from commit 4ee2de1e38a85abb89f9f313a3424c7368e12d1a)

RNHTTR committed:
Update faq.rst about missing task logs (#30717)

---------

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit f3e82b2011faaf2c48ca1258a30de342b2af2cd1)

awbush committed:
Fix backfill KeyError when try_number out of sync (#30653)

Co-authored-by: Anthony Bush <anthony.bush@reddit.com>
(cherry picked from commit 17d1f3a7bbb0fc528e7e9f082d7ada3caddcf5e1)

potiuk committed:
Upgrade PIP to 23.1.2 (#30964)

(cherry picked from commit 8b13a8bd85312b37bd485831f6b617baae5e5c57)

ephraimbuddy committed:
Update 2.6.0 release date (#30971)

(cherry picked from commit 817f846e05b2423ab407ebd24014458917f1235d)

ephraimbuddy committed:
Update incorrect config values (#30969)

(cherry picked from commit ced56ea5a783a2846094996503d84fc21a33d6ae)

pankajastro committed:
Fix `airflow providers get` command output (#30978)

If I use the `airflow provider get <provider-name> -o json/yaml` CLI command
it does not respect `-o` option.
It works perfectly fine when I pass the `--full` option also in the above command.
The issue is currently if the `--full` option is not provided we
use standard statement print to print results and not the custom AirflowConsole.
In this PR I'm replacing print with `AirflowConsole().print_as`

(cherry picked from commit 23b5b31d48a1c8fc784022c641b33b4ba837d549)

herlambang committed:
Fix unable to remove DagRun and TaskInstance with note (#30987)

* Define cascade option on note relationship within DagRun and TaskInstance model

---------

Co-authored-by: herlambang <herlambang@wartek.belajar.id>
(cherry picked from commit 0212b7c14c4ce6866d5da1ba9f25d3ecc5c2188f)

potiuk committed:
Fix missing models in API documentation. (#31021)

The Airflow models have been accidentally removed from the docs
when #28300 was implemented. The whole models documentation have
been removed accidentally, even if there were references to the
actual classes and packages used as the models package was
entirely excluded.

This PR fixes it by selectively including the models that should
be included and by linking the package indexes directly in
the public interface documentation.

(cherry picked from commit 6720a90af7d45088a6b08d8757eb258baa936afd)

ephraimbuddy committed:
Add missing changelog in 2.6.0 (#31011)

* Add missing changelog in 2.6.0

* Update RELEASE_NOTES.rst

Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>

* fixup! Add missing changelog in 2.6.0

---------

Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>
(cherry picked from commit ea18edb06f1db3728f74c757bd7cbe2eaa845172)

pierrejeambrun committed:
Fix Pool schema OpenAPI spec (#30973)

* Fix Pool schema OpenAPI spec

* Update OpenApi example

(cherry picked from commit 20b4789ee3a9346a5fd75f8db812eb7ffa5e3f10)

jedcunningham committed:
Fix some typos in the cluster policies docs (#31031)

(cherry picked from commit 366ae51e21dfb70da85962ca753e8cba8c4d3ec8)

meujin committed:
Fix problem with displaying graph (#30765)

If run_id is presented in the graph URL for example:
http://localhost:8080/dags/simple_dag/graph?run_id=scheduled__2023-03-14T12%3A47%3A00%2B00%3A00&execution_date=2023-03-14+12%3A47%3A00%2B00%3A00
query_date will be equal to base_date which set as (timezone.utcnow() + timedelta(seconds=1)).replace(microsecond=0)
And drs will contain only recent num_runs results which may not include dag_run with desirable execution_date, instead of this you will have the latest one.

(cherry picked from commit eb7b98d368a9d00ab6c6790dd85e327ed9c9e826)

strickvl committed:
Small fixes (#31015)

(cherry picked from commit 6c92efbe8b99e172fe3b585114e1924c0bb2f26b)

dvfeinblum-betterment committed:
Improve Description of dag_processing.processes Metric in Docs (#30891)

* improve description of dag_processing.processes

* clarifying meaning of "relative"

* typo

(cherry picked from commit 963717047125e96f3e704eeb3a84110d283985c2)

LipuFei committed:
Move TaskInstanceKey to a separate file (#31033)

closes: #30988

Fixes a circular import of DagRun -> TaskInstance -> Sentry ->
KubernetesExexcutor -> kubernetes_helper_functions.py (*) ->
TaskInstanceKey.

Moving TaskInstanceKey out to a separate file will break the cycle from
kubernetes_helper_functions.py -> taskinstances.py

Co-authored-by: Lipu Fei <lipu.fei@kpn.com>
(cherry picked from commit ac46902154c060246dec942f921f7670015e6031)

ephraimbuddy committed:
Fix issue in pre-importing modules in zipfile (#31061)

This fixes the scheduler crash due to reading a zipped file as source code
when trying to pre-import airflow modules from the python modules in the zip files.
The fix was to extract the python modules in the zip file and then pre-import airflow modules in them

apply suggestions from code review

(cherry picked from commit 34b6230f3c7815b8ae7e99443e45a56921059d3f)

ferruzzi committed:
Small typo in setup.py (#31049)

* small typo in comment

* found another

(cherry picked from commit 9720705d9281c8f13d96f4165fab77264306d517)

raphaelauv committed:
[DOC] README - K8S support 2.6.0 (#31085)

(cherry picked from commit 433ed2cf0e5843a4697663a5b1e4f74c872f74f6)

davidshur committed:
fix EXTRAS_DEPENDENCIES of apache-hive (#31068)

(cherry picked from commit 9e43d4aee3b86134b1b9a42f988fb9d3975dbaf7)

tatiana committed:
Add support for dynamic tasks with template fields that contain `pandas.DataFrame` (#30943)

Closes: https://github.com/astronomer/astro-sdk/issues/1911
(cherry picked from commit e126530667fcac32d324d28491bee792cb71b1d8)

mtsadler-branch committed:
Format DagBag.dagbag_report() Output (#31095)

The whitespace before `{table}`
is throwing off the headers when printing `DagBag.dagbag_report()`

(cherry picked from commit 41c87464428d8d31ba81444b3adf457bc968e11d)

herlambang committed:
Fix template error when iterating None value and fix params documentation (#31078)

* Check if form value is iterable

* Update doc params list to array

* Add test render, prevent value mutaion in js

(cherry picked from commit b8b18bd74b72edc4b40e91258fccc54cf3aff3c1)

dstandish committed:
Move interleave_timestamp_parser config to section logging (#31102)

(cherry picked from commit 94c24a1986b2a3a73c871683a555833ad244dd0d)

JCoder01 committed:
fix MappedTaskGroup import in taskinstance (#31100)

(cherry picked from commit 451b4ad8a45697947551fde711d997dd4d9cd7cf)

skytin1004 committed:
Fix typos in dev documentation (#31115)

(cherry picked from commit 4937a18b8494dfa8977e18889324222808bb27ab)

herlambang committed:
Mask task attribute on task detail view (#31125)

(cherry picked from commit ffe3a68f9ada2d9d35333d6a32eac2b6ac9c70d6)

ephraimbuddy committed:
Update version to 2.6.1

ephraimbuddy committed:
Update RELEASE_NOTES.rst

potiuk reviewed (approved):
LGTM. I have one  doubt about https://github.com/apache/airflow/pull/30352 - it has new configuraiton param on one hand, but it does solve a problem where  redis sentinel configuration was broken. So I would be in getting it in as it looks pretty safe/

potiuk reviewed (approved):
LGTM. I have one  doubt about https://github.com/apache/airflow/pull/30352 - it has new configuraiton param on one hand, but it does solve a problem where  redis sentinel configuration was broken. So I would be in getting it in as it looks pretty safe/

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+Misc/Internal
+"""""""""""""
+- Present affected task instances as table (#30633)
+- Fix missing models in API documentation. (#31021)
```
```suggestion
- Show task instances affected by clearing in a table (#30633)
- Fix missing models in API documentation (#31021)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+""""""""""""""""
+- Improve Description of dag_processing.processes Metric in Docs (#30891)
+- Improve Quick Start instructions (#30820)
+- Update faq.rst about missing task logs (#30717)
```
```suggestion
- Add section about missing task logs to the FAQ (#30717)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+
+Doc only changes
+""""""""""""""""
+- Improve Description of dag_processing.processes Metric in Docs (#30891)
```
```suggestion
- Improve description of the ``dag_processing.processes`` metric (#30891)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+- Improve Description of dag_processing.processes Metric in Docs (#30891)
+- Improve Quick Start instructions (#30820)
+- Update faq.rst about missing task logs (#30717)
+- docker compose doc changes (#30662)
```
```suggestion
- Mount the `config` directory in docker compose (#30662)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+- Fix backfill KeyError when try_number out of sync (#30653)
+- Re-enable clear on TaskInstanceModelView for role User (#30415)
+- Fix DAG.access_control can't sync when clean access_control (#30340)
+- Prevent DagRun's ``start_date`` from reset (#30124) (#30125)
```
```suggestion
- Prevent DagRun's ``state`` and ``start_date`` from being reset when clearing a task in a running DagRun (#30125)
```

raphaelauv commented on README.md:
```diff
@@ -87,15 +87,15 @@ Airflow is not a streaming solution, but it is often used to process real-time d
+|---------------------|------------------------------|------------------------------------|
+| Python              | 3.7, 3.8, 3.9, 3.10          | 3.7, 3.8, 3.9, 3.10                |
+| Platform            | AMD64/ARM64(\*)              | AMD64/ARM64(\*)                    |
+| Kubernetes          | 1.23, 1.24, 1.25, 1.26       | 1.21, 1.22, 1.23, 1.24, 1.25, 1.26 |
```
you should cherry-pick this commit https://github.com/apache/airflow/pull/31085

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+- Fix mapped states case (#30916)
+- Fix problem with displaying graph (#30765)
+- Fix backfill KeyError when try_number out of sync (#30653)
+- Re-enable clear on TaskInstanceModelView for role User (#30415)
```
```suggestion
- Re-enable clear and setting state in the TaskInstance UI (#30415)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+- Fix problem with displaying graph (#30765)
+- Fix backfill KeyError when try_number out of sync (#30653)
+- Re-enable clear on TaskInstanceModelView for role User (#30415)
+- Fix DAG.access_control can't sync when clean access_control (#30340)
```
I have a concern with this change. I think it fixes one issue (empty access_control isn't respected), but creates another (it wipes out any rights not set by access_control initially).

ephraimbuddy replied:
Then I will drop it

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+- Include node height/width in center-on-task logic (#30924)
+- Remove print from dag trigger command (#30921)
+- Improve task group UI in new graph (#30918)
+- Fix mapped states case (#30916)
```
```suggestion
- Fix mapped states in grid view (#30916)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+- Fix unable to remove DagRun and TaskInstance with note (#30987)
+- Fix ``airflow providers get`` command output (#30978)
+- Fix Pool schema OpenAPI spec (#30973)
+- Update incorrect config values (#30969)
```
This feels like a docs-only change really. We should update the wording too.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+- Move TaskInstanceKey to a separate file (#31033)
+- Fix unable to remove DagRun and TaskInstance with note (#30987)
+- Fix ``airflow providers get`` command output (#30978)
+- Fix Pool schema OpenAPI spec (#30973)
```
```suggestion
- Fix Pool schema in the OpenAPI spec (#30973)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+- fix EXTRAS_DEPENDENCIES of apache-hive (#31068)
+- Fix issue in pre-importing modules in zipfile (#31061)
+- Move TaskInstanceKey to a separate file (#31033)
+- Fix unable to remove DagRun and TaskInstance with note (#30987)
```
```suggestion
- Fix deleting DagRuns and TaskInstances that have a note (#30987)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+- Fix template error when iterating None value and fix params documentation (#31078)
+- fix EXTRAS_DEPENDENCIES of apache-hive (#31068)
+- Fix issue in pre-importing modules in zipfile (#31061)
+- Move TaskInstanceKey to a separate file (#31033)
```
```suggestion
- Move TaskInstanceKey to a separate file to fix circular import (#31033)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+- Format DagBag.dagbag_report() Output (#31095)
+- Fix template error when iterating None value and fix params documentation (#31078)
+- fix EXTRAS_DEPENDENCIES of apache-hive (#31068)
+- Fix issue in pre-importing modules in zipfile (#31061)
```
```suggestion
- Fix issue with zipfiles in DAGs folder when pre-importing Airflow modules (#31061)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+- fix MappedTaskGroup import in taskinstance (#31100)
+- Format DagBag.dagbag_report() Output (#31095)
+- Fix template error when iterating None value and fix params documentation (#31078)
+- fix EXTRAS_DEPENDENCIES of apache-hive (#31068)
```
```suggestion
- Fix ``apache-hive`` extra so it installs the correct package (#31068)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+Bug Fixes
+^^^^^^^^^
+- Move interleave_timestamp_parser config to section logging (#31102)
+- fix MappedTaskGroup import in taskinstance (#31100)
```
```suggestion
- Fix ``MappedTaskGroup`` import in taskinstance file (#31100)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,53 @@
+
+Bug Fixes
+^^^^^^^^^
+- Move interleave_timestamp_parser config to section logging (#31102)
```
```suggestion
- Move ``interleave_timestamp_parser`` config to the logging section (#31102)
```

potiuk reviewed (changes requested):
Also - should not we also include that one? https://github.com/apache/airflow/pull/31101 ? I think that one is pretty important to get merged @dstandish @ephraimbuddy ? - a number of people have hard time finding no logs in their tasks.

dstandish committed:
Ensure that we check worker for served logs if no local or remote logs found (#31101)

Generally speaking when a task is done we should not read from logs server, because we assume for log persistence that users will either set up shared drive or enable remote logging.  But if they don't do that, and therefore we don't find remote or local logs, we'll check worker for served logs as a fallback.

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit 672ee7f0e175dd7edb041218850d0cd556d62106)

ashb committed:
Remove worrying log message about redaction from the OpenLineage plugin (#31149)

The current published plugin calls `_redact_all` directly, so when we
added the `max_depth` parameter in 2.6 (precisely so it didn't need to
call this private function directly!) we broke that.

This change leads to some worrying logs appearing in task logs for
anyone with the plugin installed:

```
[2023-05-05, 11:56:17 BST] {utils.py:490} WARNING - Unable to redact []Error was: TypeError: SecretsMasker._redact_all() missing 1 required positional argument: 'max_depth'
[2023-05-05, 11:56:17 BST] {utils.py:490} WARNING - Unable to redact []Error was: TypeError: SecretsMasker._redact_all() missing 1 required positional argument: 'max_depth'
[2023-05-05, 11:56:17 BST] {utils.py:490} WARNING - Unable to redact NoneError was: TypeError: SecretsMasker._redact_all() missing 1 required positional argument: 'max_depth'
```

This patch makes that previous change backwards compatible for that with
no change in behaviour for new code.

(cherry picked from commit 25e315995e53afef83fe310426c29e3bea14cd27)

jedcunningham committed:
Ensure the KPO runs pod mutation hooks correctly (#31173)

If you run a task via the cli with `cfg_path`, we also run
`configure_vars` again so that the globals built from the config file
can be updated. We were also initializing pluggy in that function,
however, which was effectively removing any user provided hooks.

We will move the initializing of pluggy out of that function, making it
safe to run again by the task command.

This primarily impacted KPOs, as they wouldn't have been passed through any `pod_mutation_hooks`.

(cherry picked from commit 71c26276bcd3ddd5377d620e6b8baef30b72eaa0)

dstandish committed:
Fix section name reference for api_client_retry_configuration (#31174)

(cherry picked from commit 289065285ca54c3d7c2a5a4f8bf86efeef5d1507)

AmFlint committed:
Make sure that DAG processor job row has filed value in `job_type` column (#31182)

* fix dag processor job_runner, add missing job_type

* remove uneeded spaces

* move job_type check to base_runner

* fix test mock job runner pass job argument to init

(cherry picked from commit 420a9b1581bf88ca8b8ed18df00d63bb6900547e)

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,7 +21,55 @@
 .. towncrier release notes start
 
-Airflow 2.6.0 (2023-04-28)
+Airflow 2.6.1 (2023-05-12)
```
```suggestion
Airflow 2.6.1 (2023-05-15)
```

Let's do the 15th so we aren't releasing over the weekend again.

dstandish committed:
Fix timestamp parse failure for k8s executor pod tailing (#31175)

(cherry picked from commit 86d62d3c0398d0d842c9368c5832de92bc6ec4c5)
